controllers/topics/index.js

```diff
@@ -337,12 +337,18 @@ const followTopicV2 = async (req, res) => {
 
 const _topicAutoMessage = async (user_id, topic_id, detailTokenUser) => {
   try {
-    const communityMessageFormat = await CommunityMessageFormat.findOne({
+    const communityMessageFormats = await CommunityMessageFormat.findAll({
       where: {topic_id, status: '1'}
     });
-
-    if (communityMessageFormat && !detailTokenUser.is_anonymous) {
-      followTopicServiceQueue(user_id, topic_id);
+    if (communityMessageFormats && !detailTokenUser.is_anonymous) {
+      for (const format of communityMessageFormats) {
+        followTopicServiceQueue(
+          user_id,
+          topic_id,
+          format.community_message_format_id,
+          format.delay_time
+        );
+      }
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The error handling in this section could be improved. Currently, if an error occurs within the `followTopicServiceQueue` function, it will stop the execution of the entire loop. Consider wrapping the call to `followTopicServiceQueue` in a try-catch block to handle errors for each iteration separately. This way, even if one call fails, the loop can continue with the next item.

```diff
    for (const format of communityMessageFormats) {
+     try {
        followTopicServiceQueue(
          user_id,
          topic_id,
          format.community_message_format_id,
          format.delay_time
        );
+     } catch (error) {
+       console.error(`Error processing format ${format.community_message_format_id}:`, error);
+     }
    }
```

Also, using `console.log` for error logging is not recommended for production code. It's better to use a dedicated logging library that provides more flexibility and control over log levels, formats, and destinations.


<!-- This is an auto-generated comment by OSS CodeRabbit -->